cmd/package_managers.go

```diff
@@ -108,6 +113,51 @@ func fetchGitRepositoryFromNPM(packageName string, packageManager pmc.Client) (s
 	return v.Objects[0].Package.Links.Repository, nil
 }
 
+func repoIfValid(url string) string {
+	match := _GITHUB_DOMAIN_REGEXP.FindStringSubmatch(url)
+	if len(match) >= 3 {
+		return fmt.Sprintf("https://github.com/%s/%s", match[1], match[2])
+	}
+
+	match = _GITHUB_SUBDOMAIN_REGEXP.FindStringSubmatch(url)
+	if len(match) >= 3 {
+		return fmt.Sprintf("https://github.com/%s/%s", match[1], match[2])
+	}
+
+	match = _GITLAB_DOMAIN_REGEXP.FindStringSubmatch(url)
+	if len(match) >= 3 {
+		return fmt.Sprintf("https://gitlab.com/%s/%s", match[1], match[2])
+	}
+	return ""
+}
+
+func findGitRepositoryInPYPIResponse(packageName string, response io.Reader) (string, error) {
+	v := &pypiSearchResults{}
+	err := json.NewDecoder(response).Decode(v)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("failed to parse pypi package json: %v", err))
+	}
```

Comment:
consider using `fmt.Errorf` and the `%w` directive

Reply:
Note: the author is matching the idioms in the rest of the file.

So probably something we (the maintainers) would need to fix more broadly. `ErrScorecardInternal` is something that should really be a wrapper error type which supports inner errors, but we're stuck with what we have now for backwards compatibility, and we're not using go1.20 yet for multi-errors.

Reply:
Ack leaving as is for now